watchdog/src/lib.rs

```diff
@@ -23,9 +26,17 @@ thread_local! {
     static BLOCK_INFO_DATA: RwLock<HashMap<BitcoinBlockApi, BlockInfo>> = RwLock::new(HashMap::new());
 }
 
+static START: Once = Once::new();
+
 /// This function is called when the canister is created.
 #[ic_cdk_macros::init]
 fn init() {
+    // Setup the stdlib hooks.
+    START.call_once(|| {
```

Comment:
I don't think we need to use the `Once` primitive here. The canister's `init` method will only be invoked once at canister initialization anyway. We should also setup the hook in the `post_upgrade` method (so if the canister is upgraded we don't lose it).

Reply:
Done. `init` method is called from `post_upgrade` method so it's handled automatically.